src/Sylius/Bundle/ApiBundle/Context/RequestBasedChannelContext.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Context;
+
+use ApiPlatform\Core\Metadata\Resource\Factory\ResourceMetadataFactoryInterface;
+use Sylius\Bundle\ApiBundle\Metadata\Resource\OperationCheckerTrait;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Channel\Context\ChannelNotFoundException;
+use Sylius\Component\Channel\Model\ChannelInterface;
+use Sylius\Component\Channel\Repository\ChannelRepositoryInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+final class RequestBasedChannelContext implements ChannelContextInterface
```

Comment:
This class is not needed. We will not provide a channel in the path attribute. Let's re-use what we have in core instead.